src/objects/file.py

```diff
@@ -11,93 +12,194 @@
+
+        :param binary: file in binary representation
+        '''
         self.binary = make_bytes(binary)
         self.sha256 = get_sha256(self.binary)
         self.size = len(self.binary)
         self._uid = create_uid(binary)
 
-    def set_name(self, name):
-        self.file_name = make_unicode_string(name)
-
-    def set_file_path(self, file_path):
-        self.file_path = file_path
-        if self.binary is None:
-            self.create_from_file(file_path)
-        if self.file_name is None:
-            self.set_name(os.path.basename(file_path))
-
-    def get_uid(self):
-        logging.warning('Deprecation warning: "get_uid()" was replaced by "uid" and will be removed in a future update')
-        return self.uid
+    def create_binary_from_path(self) -> None:
+        if self.file_path is not None:
+            if self.binary is None:
+                self._create_from_file(self.file_path)
+            if self.file_name is None:
+                self.file_name = make_unicode_string(Path(self.file_path).name)
 
     @property
-    def uid(self):
+    def uid(self) -> str:
+        '''
+        Unique identifier of this file.
+        Consisting of the file's sha256 hash and it's length in the form `hash_length`.
+
+        :return: uid of this file.
+        '''
         if self._uid is None and self.binary is not None:
             self._uid = create_uid(self.binary)
         return self._uid
 
     @uid.setter
-    def uid(self, new_uid):
+    def uid(self, new_uid: str):
         if self._uid is not None:
             logging.warning('uid overwrite: Uid might not be related to binary data anymore: {} -> {}'.format(self._uid, new_uid))
         self._uid = new_uid
 
-    def get_hid(self, root_uid=None):
+    def get_hid(self, root_uid: str = None) -> str:
         '''
-        return a human readable identifier
+        Get a human readable identifier for the given file.
+        This usually is the file name for extracted files.
+        As files can have different names across occurences, uid of a specific root object can be specified.
+
+        :param root_uid: (Optional) root uid to base HID on.
+        :return: String presenting human readable identifier for file.
         '''
         if root_uid is None:
             root_uid = self.get_root_uid()
         virtual_path = self.get_virtual_paths_for_one_uid(root_uid=root_uid)[0]
         return get_top_of_virtual_path(virtual_path)
 
-    def create_from_file(self, file_path):
+    def _create_from_file(self, file_path: str):
         self.set_binary(get_binary_from_file(file_path))
-        self.set_file_path(file_path)
+        self.create_binary_from_path()
+
+    def add_included_file(self, file_object) -> None:
+        '''
+        This functions adds a file to this objects list of included files.
+        The function also takes care of a number of meta information for the child object:
```

Comment:
as "information" is not countable, maybe
```suggestion
        The function also takes care of a number of meta information fields for the child object:
```
would be better?